python/cudf/cudf/core/dataframe.py

```diff
@@ -179,6 +179,9 @@ def __init__(self, data=None, index=None, columns=None, dtype=None):
             self._index = as_index(index)
             return None
 
+        if isinstance(data, pd.DataFrame):
+            data = self.from_pandas(data)
```

Comment:
Can we move this to below the `isinstance(data, DataFrame)` branch? Just to make handling already on GPU data that itsy bitsy faster by avoiding an additional `isinstance` check 😄

Reply:
done 👍